File: Competitive Programming 3/Section 1.4/Games (Others), Easier/340 - Master-Mind Hints/Python/main.py
def main():
    cs=1
    while True:
        N = int(input())
        if not N:
            break
        print('Game '+str(cs)+':')
        s = list(map(int,input().split()))
        while True:
            g = list(map(int,input().split()))
            if not g[0]:
                break
            strong,weak=0,0
            numbersInS = {}
            for i in range(N):
                if s[i] == g[i]:
                    strong+=1
                else: 
                    if s[i] in numbersInS:
                        numbersInS[s[i]]+=1
                    else:
                        numbersInS[s[i]]=1
            for i in range(N):
                if s[i] != g[i] and g[i] in numbersInS:
                    weak+=1
                    if numbersInS[g[i]]==1:
                        del numbersInS[g[i]]
                    else:
                        numbersInS[g[i]]-=1
            print('    ('+str(strong)+','+str(weak)+')')
        cs+=1
main()

File: Competitive Programming 3/Section 1.4/Games (Others), Easier/340 - Master-Mind Hints/C++/main.cpp
#include<bits/stdc++.h>

#define lim 1000

using namespace std;

int N,s[lim],g[lim];

bool readG(){
    for(int i = 0 ; i < N ; i ++)
        cin >> g[i];
    return g[0];
}

int main(){
    ios_base::sync_with_stdio(0);cin.tie(0);
    for(int cs=1;cin>>N,N;cs++){
        cout << "Game " << cs << ":\n";
        for(int i = 0 ; i < N ; i ++)
            cin >> s[i];
        while(readG()){
            int strong=0,weak=0;
            multiset<int>numbersInS;
            for(int i = 0 ; i < N ; i++)
                if(s[i]==g[i])
                    strong++;
                else
                    numbersInS.insert(s[i]);
            for(int i = 0 ; i < N ; i ++){
                auto it = numbersInS.find(g[i]);
                if(s[i]!=g[i]&&it!=numbersInS.end()){
                    weak++;
                    numbersInS.erase(it);
                }
            }
            cout << "    (" << strong << "," << weak << ")\n";
        }
    }
    return 0;
}